### ISSUE TITLE
SIGSEGV on world start

### ISSUE THREAD

**Issue by RenechCDDA** [CleverRaven/Cataclysm-DDA#68500]
### Describe the bug

Try to start game ---> instant crash

### Attach save file

N/A

### Steps to reproduce

Was reproducing #68422. New world, default settings, scenario: Challenge - Hunted. Apartment rooftops start.

~~Was not able to reproduce after a few tries, but~~ the crash log is similar to another recent issue (see additional context)

### Expected behavior

Game does not crash on start

### Versions and configuration

5621850

Windows 10

MSVC compiled (VS2019)

### Additional context

[crash.log](https://github.com/CleverRaven/Cataclysm-DDA/files/12806289/crash.log)

This looks like a variation on the #66427 crash?

**Comment by snipercup** [comment]
My game also crashed when starting a new world. 
1. Make new world. Default settings + magiclysm + tameable wildlife
2. From the menu, select "play now! - Random character"

Same version and OS as OP

**Comment by RenechCDDA** [comment]
Marking confirmed based on reproductions in the development discord.

**Comment by RenechCDDA** [comment]
This also occurs when long-range teleporting, both debug (dev discord reproductions) and recently when I was testing #69364

**Comment by BrettDong** [comment]
I also encountered a similar crash on starting a new world.

Stack trace, hopefully helpful for someone who are familiar with mapgen:

```
Process 64309 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=2, address=0x1285a88cf4)
    frame #0: 0x0000000100a7bd68 cataclysm-tiles`map::pl_sees(tripoint const&, int) const [inlined] std::__1::__less<float, float>::operator()[abi:v160006](this=<unavailable>, __x=<unavailable>, __y=<unavailable>) const at comp.h:47:67 [opt]
   44  	struct __less<_T1, _T1>
   45  	{
   46  	    _LIBCPP_INLINE_VISIBILITY _LIBCPP_CONSTEXPR_SINCE_CXX14
-> 47  	    bool operator()(const _T1& __x, const _T1& __y) const {return __x < __y;}
   48  	};
   49  	
   50  	template <class _T1>
Target 0: (cataclysm-tiles) stopped.
warning: cataclysm-tiles was compiled with optimization - stepping may behave oddly; variables may not be available.
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=2, address=0x1285a88cf4)
  * frame #0: 0x0000000100a7bd68 cataclysm-tiles`map::pl_sees(tripoint const&, int) const [inlined] std::__1::__less<float, float>::operator()[abi:v160006](this=<unavailable>, __x=<unavailable>, __y=<unavailable>) const at comp.h:47:67 [opt]
    frame #1: 0x0000000100a7bd68 cataclysm-tiles`map::pl_sees(tripoint const&, int) const at max_element.h:33:17 [opt]
    frame #2: 0x0000000100a7bd5c cataclysm-tiles`map::pl_sees(tripoint const&, int) const [inlined] float const* std::__1::max_element[abi:v160006]<float const*, std::__1::__less<float, float>>(__first=<unavailable>, __last=<unavailable>, __comp=<unavailable>) at max_element.h:43:12 [opt]
    frame #3: 0x0000000100a7bd5c cataclysm-tiles`map::pl_sees(tripoint const&, int) const [inlined] float const* std::__1::max_element[abi:v160006]<float const*>(__first=<unavailable>, __last=<unavailable>) at max_element.h:51:12 [opt]
    frame #4: 0x0000000100a7bd5c cataclysm-tiles`map::pl_sees(tripoint const&, int) const [inlined] four_quadrants::max(this=<unavailable>) const at shadowcasting.h:57:17 [opt]
    frame #5: 0x0000000100a7bd5c cataclysm-tiles`map::pl_sees(this=<unavailable>, t=0x000000016fdf7800, max_range=<unavailable>) const at lightmap.cpp:804:98 [opt]
    frame #6: 0x00000001002ba3f4 cataclysm-tiles`Character::sees(this=0x0000000137a22a00, t=0x000000016fdf7800, (null)=<unavailable>, (null)=<unavailable>) const at character.cpp:10966:44 [opt]
    frame #7: 0x0000000100aecd38 cataclysm-tiles`ter_furn_transform::transform(this=0x00000001270d9408, m=0x000000016fdf82e8, location=0x000000016fdf7800) const at magic_ter_fur_transform.cpp:282:17 [opt]
    frame #8: 0x0000000100c3b4e0 cataclysm-tiles`jmapgen_ter_furn_transform::apply(this=<unavailable>, dat=0x000000016fdf7980, x=<unavailable>, y=0x000000016fdf7868, (null)=<unavailable>) const at mapgen.cpp:3005:24 [opt]
    frame #9: 0x0000000100bc3eec cataclysm-tiles`jmapgen_objects::apply(this=0x00000001261de028, dat=0x000000016fdf7980, phase=<unavailable>, offset=0x000000016fdf7b08, context="nested mapgen damaged_door_nested in mapgen house_31") const at mapgen.cpp:5237:18 [opt]
    frame #10: 0x0000000100bc3200 cataclysm-tiles`apply_mapgen_in_phases(md=0x000000016fdf7980, setmap_points=size=0, objects=0x00000001261de028, offset=0x000000016fdf7b08, context="nested mapgen damaged_door_nested in mapgen house_31", verify=false) at mapgen.cpp:5107:17 [opt]
    frame #11: 0x0000000100bc3cd4 cataclysm-tiles`mapgen_function_json_nested::nest(this=0x00000001261ddf48, md=0x000000016fdf7c80, offset=0x000000016fdf7b08, outer_context="mapgen house_31") const at mapgen.cpp:5197:5 [opt]
    frame #12: 0x0000000100c25c64 cataclysm-tiles`jmapgen_nested::apply(this=<unavailable>, dat=0x000000016fdf7c80, x=0x000000016fdf7b60, y=0x000000016fdf7b68, context="mapgen house_31") const at mapgen.cpp:3746:23 [opt]
    frame #13: 0x0000000100bc3eec cataclysm-tiles`jmapgen_objects::apply(this=0x0000000124a94328, dat=0x000000016fdf7c80, phase=<unavailable>, offset=0x0000000101889978, context="mapgen house_31") const at mapgen.cpp:5237:18 [opt]
    frame #14: 0x0000000100bc3200 cataclysm-tiles`apply_mapgen_in_phases(md=0x000000016fdf7c80, setmap_points=size=0, objects=0x0000000124a94328, offset=0x0000000101889978, context="mapgen house_31", verify=false) at mapgen.cpp:5107:17 [opt]
    frame #15: 0x0000000100bc2e78 cataclysm-tiles`mapgen_function_json::generate(this=0x0000000124a94248, md=0x000000016fdf80c8) at mapgen.cpp:5166:5 [opt]
    frame #16: 0x0000000100bd8144 cataclysm-tiles`mapgen_basic_container::generate(this=<unavailable>, dat=0x000000016fdf80c8, hardcoded_weight=0) at mapgen.cpp:350:23 [opt]
    frame #17: 0x0000000100bc3f68 cataclysm-tiles`run_mapgen_func(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, mapgendata&) [inlined] mapgen_factory::generate(this=<unavailable>, dat=<unavailable>, key=<unavailable>, hardcoded_weight=0) at mapgen.cpp:483:33 [opt] [artificial]
    frame #18: 0x0000000100ba6148 cataclysm-tiles`map::draw_map(this=0x000000016fdf82e8, dat=0x000000016fdf80c8) at mapgen.cpp:5262:28 [opt]
    frame #19: 0x0000000100ba59a0 cataclysm-tiles`map::generate(this=0x000000016fdf82e8, p=0x000000016fdf85b8, when=0x0000000101cbad78) at mapgen.cpp:212:5 [opt]
    frame #20: 0x0000000100b344e8 cataclysm-tiles`map::loadn(this=0x0000000136e857b0, grid=0x000000016fdf8650, update_vehicles=true) at map.cpp:8041:21 [opt]
    frame #21: 0x0000000100b32584 cataclysm-tiles`map::loadn(this=0x0000000136e857b0, grid=0x000000016fdf86a8, update_vehicles=true) at map.cpp:8109:13 [opt]
    frame #22: 0x0000000100b32418 cataclysm-tiles`map::load(this=0x0000000136e857b0, w=<unavailable>, update_vehicle=true, pump_events=true) at map.cpp:7684:13 [opt]
    frame #23: 0x00000001006c95b0 cataclysm-tiles`game::start_game() [inlined] game::load_map(this=0x000000013799b200, pos_sm=0x000000016fdf8820, pump_events=true) at game.cpp:823:7 [opt]
    frame #24: 0x00000001006c959c cataclysm-tiles`game::start_game(this=0x000000013799b200) at game.cpp:894:5 [opt]
    frame #25: 0x0000000100b0a4c8 cataclysm-tiles`main_menu::new_character_tab(this=0x000000016fdfeac0) at main_menu.cpp:999:17 [opt]
    frame #26: 0x0000000100b0766c cataclysm-tiles`main_menu::opening_screen(this=0x000000016fdfeac0) at main_menu.cpp:863:29 [opt]
    frame #27: 0x0000000100afc664 cataclysm-tiles`main(argc=<unavailable>, argv=0x000000016fdff258) at main.cpp:791:19 [opt]
    frame #28: 0x00000001801a90e0 dyld`start + 2360
(lldb) frame select 6
frame #6: 0x00000001002ba3f4 cataclysm-tiles`Character::sees(this=0x0000000137a22a00, t=0x000000016fdf7800, (null)=<unavailable>, (null)=<unavailable>) const at character.cpp:10966:44 [opt]
   10963	bool Character::sees( const tripoint &t, bool, int ) const
   10964	{
   10965	    const int wanted_range = rl_dist( pos(), t );
-> 10966	    bool can_see = is_avatar() ? get_map().pl_sees( t, std::min( sight_max, wanted_range ) ) :
   10967	                   Creature::sees( t );
   10968	    // Clairvoyance is now pretty cheap, so we can check it early
   10969	    if( wanted_range < MAX_CLAIRVOYANCE && wanted_range < clairvoyance() ) {
(lldb) p t
(const tripoint &) 0x000000016fdf7800 (x = 12, y = 14, z = 0)
(lldb) p sight_max
(int) 9999
(lldb) p wanted_range
(const int) 3483
(lldb) p pos()
(tripoint)  (x = -3280, y = -1010, z = -500)
```

**Comment by Inglonias** [comment]
Looks like I also can reproduce this

**Comment by RenechCDDA** [comment]
Resolved by #69916 / #69948

**Comment by RenechCDDA** [comment]
Resolved by #69916 / #69948
